### ISSUE TITLE
Does Nightmare ship with its own headless browser or do I need to install one?

### ISSUE THREAD

**Issue by reselbob** [segmentio/nightmare#1472]
I am trying to save time.

I have some GUI tests that are driven by Nightmare, under `npm test` that I run under a BitBucket-Pipeline deployment. Thus, it's a pristine environment.

The tests run fine on my development machine. But, I have Chrome and Safari installed on this system.

I was assuming that Nightmare shipped with its own headless browser as part its package. But, I am getting a timeout error when I run the tests under Bitbucket-Pipeline. So, I am thinking that the GUI tests might be looking around for a brower (headless) that's not installed. And, that it does not find one, the test times out.

So, I ask does Nightmare ship with its own headless browser or do I need to install one?

**Comment by PhmNgocNghia** [comment]
Nightmare use electron as it's browser, also it's dependency you don't need to install anything. `nightmare = Nightmare({ show: false })` behave just like headless browser in my opinion. If you chrome as headless browser you may consider use puppeteer.

**Comment by val7000** [comment]
@PhmNgocNghia So, do you consider puppeteer as something better or there are any alternatives?

**Comment by PhmNgocNghia** [comment]
No, but you can try anything but not nightmare since it's not active maintain anymore. just look at `Latest commit 25467e5  on Mar 29`. I recommend to use `Chromeless` or `Puppeter`. The different between two is:
* With chromeless: you can't inject javascript into page (jquery lover ?)
* With puppeter: you can't make chain method so instead you write
```js
page
  .lorem(0)
  .ispum(1)
```
You write : 
```js
page.lorem(0)
page.ispum(1)
```

Me ? I use puppeter because it's official headless chrome library by Google.

**Comment by val7000** [comment]
Agree, chain methods are practical. But more valuable are maintence of library and no bugs. Puppeter is favorite.
